quickwit/quickwit-metastore/src/metastore/file_backed_metastore/mod.rs

```diff
@@ -497,8 +497,8 @@ impl Metastore for FileBackedMetastore {
         .await
     }
 
-    fn uri(&self) -> &Uri {
-        self.storage.uri()
+    fn uri(&self) -> Uri {
```

Comment:
Why not let the caller decide whether they want to clone or not instead of changing the signature?

Reply:
Yes, I was going to add a comment on this because I was forced to do it :/

I did not manage to make it work with a dynamic Uri like the one in the grpc metastore